opt/scopeserver/dataserver/utils/loom.py

```diff
@@ -416,6 +417,43 @@ def add_user_clustering(self, request) -> Tuple[bool, str]:
             logger.debug("Failure")
             return (False, "Updated clusterings do not exist in file. Write error.")
 
+    def get_cluster_overlaps(self, cell_idxs):
+        """Calculate clusters overlapped by provided cell indices
+
+        Args:
+            cell_idxs (list[int]): A list of cell indices within the loom file
+
+        Returns:
+            list[dict]: A list containing a dict with the overlaps information
+        """
+
+        metadata = self.get_meta_data()
+        cluster_overlap_data = []
+        if not self.has_md_clusterings():
+            return []
+        for clustering_meta in metadata["clusterings"]:
+            clustering_name = clustering_meta["name"]
+            clustering_id = clustering_meta["id"]
+            clustering = self.get_clustering_by_id(clustering_id)
+            counts = Counter(clustering[cell_idxs])
+            all_counts = Counter(clustering)
+            for cluster_id in counts:
```

Comment:
This loop could be a separate function and implemented with a list comprehension?

Reply:
This is a bit much to squeeze into a comprehension no? I'm also not sure if I see the added benefit of a new function either, the resulting list is going to use all of the parsed information from above, which means passing it all along. Do you have a specific reason to want to pull it out/put it in a comprehension?